fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1879,10 +1886,12 @@ def update_p_data(custom_output_tensor: Optional[torch.Tensor] = None) -> None:
                     # Fill output_tensor with (p.data for each shard in self.world_size)
                     if hasattr(dist, "_all_gather_base") and enable_nccl_base_collectives:
                         # New version of PyTorch has all_gather_base, which is faster than chunk and then all_gather.
-                        dist._all_gather_base(output_tensor, p_data, group=self.process_group)
+                        dist._all_gather_base(
+                            output_tensor, p_data, group=self.process_group["all_gather_group"], async_op=False
```

Comment:
We can probably add 2 internal variables like self._async_all_gather and self._async_reduce_scatter and use those variable instead of True/False values directly in this line and line 1894. This is less likely to cause bugs where we change one to be async and leaving another to be not async.